File: EjePersistenciaObj3/src/main/java/com/mycompany/ejepersistenciaobj3/EjePersistenciaObj3.java
package com.mycompany.ejepersistenciaobj3;
/**
 *
 * @author Vladimir Ohara
 */
public class EjePersistenciaObj3 {

    public static void main(String[] args) {
       ArchivoCliente archivo = new ArchivoCliente("clientes.txt");

        Cliente c1 = new Cliente(1, "Ana", 123456);
        Cliente c2 = new Cliente(2, "Luis", 987651);
        Cliente c3 = new Cliente(3, "Carla", 111222);

        archivo.guardaCliente(c1);
        archivo.guardaCliente(c2);
        archivo.guardaCliente(c3);

        Cliente buscado = archivo.buscarCliente(2);
        if (buscado != null) {
            System.out.println("Cliente encontrado por ID:");
            System.out.println(buscado);
        } else {
            System.out.println("Cliente no encontrado.");
        }

        Cliente buscadoTel = archivo.buscarCelularCliente(3);
        if (buscadoTel != null) {
            System.out.println("Datos del cliente con número:");
            System.out.println(buscadoTel);
        } else {
            System.out.println("Cliente no encontrado.");
        }
    }
}

File: EjePersistenciaObj3/src/main/java/com/mycompany/ejepersistenciaobj3/EjePersistenciaObj3.py
class Cliente:
    def __init__(self, id, nombre, telefono):
        self.id = id
        self.nombre = nombre
        self.telefono = telefono

    def __str__(self):
        return f"ID: {self.id}, Nombre: {self.nombre}, Teléfono: {self.telefono}"
class ArchivoCliente:
    def __init__(self, nomA):
        self.nomA = nomA
        self.clientes = []

    def crearArchivo(self):
        self.clientes = []

    def guardaCliente(self, cliente):
        self.clientes.append(cliente)

    def buscarCliente(self, c):
        for cli in self.clientes:
            if cli.id == c:
                return cli
        return None

    def buscarCelularCliente(self, c):
        return self.buscarCliente(c)
archivo = ArchivoCliente("clientes.txt")

c1 = Cliente(1, "Ana", 123456789)
c2 = Cliente(2, "Luis", 987654321)
c3 = Cliente(3, "Carla", 111222333)

archivo.guardaCliente(c1)
archivo.guardaCliente(c2)
archivo.guardaCliente(c3)

cliente_buscado = archivo.buscarCliente(2)
if cliente_buscado:
    print("Cliente encontrado por ID:")
    print(cliente_buscado)
else:
    print("Cliente no encontrado.")

cliente_tel = archivo.buscarCelularCliente(3)
if cliente_tel:
    print("Datos del cliente con número:")
    print(cliente_tel)
else:
    print("Cliente no encontrado.")
